tests/tensor_search/test_create_index.py

```diff
@@ -324,24 +324,105 @@ def test_set_number_of_shards(self):
         assert intended_shard_count == int(resp.json()[self.index_name_1]['settings']['index']['number_of_shards'])
 
     def test_set_number_of_replicas(self):
-        """ does it work if other params are filled?"""
         intended_replicas_count = 4
-        res_0 = tensor_search.create_vector_index(
+        from marqo.tensor_search.models.settings_object import settings_schema
+        with patch.dict(settings_schema["properties"][NsField.number_of_replicas], maximum=10):
+            res_0 = tensor_search.create_vector_index(
+                index_name=self.index_name_1, config=self.config,
+                index_settings={
+                    "index_defaults": {
+                        "treat_urls_and_pointers_as_images": True,
+                        "model": "ViT-B/32",
+                    },
+                    NsField.number_of_replicas: intended_replicas_count
+                }
+            )
+        resp = requests.get(
+            url=self.authorized_url + f"/{self.index_name_1}",
+            headers=self.generic_header,
+            verify=False
+        )
+        assert intended_replicas_count == int(resp.json()[self.index_name_1]['settings']['index']['number_of_replicas'])
+
+    def test_configurable_max_number_of_replicas(self):
+        maximum_number_of_replicas = 5
+        large_intended_replicas_count = 10
+        small_intended_replicas_count = 3
+        from marqo.tensor_search.models.settings_object import settings_schema
+
+        with patch.dict(settings_schema["properties"][NsField.number_of_replicas], maximum=maximum_number_of_replicas):
+            # a large value exceeding limits should not work
```

Comment:
what about a test where input number exactly equals the maximum? (`NsField.number_of_replicas: 5`)

Reply:
in this test case, there should be no `InvalidArgError`